nodejs/awsx/ec2/vpc.ts

```diff
@@ -504,23 +504,31 @@ utils.Capture(Vpc.prototype).initializeVpcArgs.doNotCapture = true;
 async function getAvailabilityZones(
         parent: pulumi.Resource | undefined,
         provider: pulumi.ProviderResource | undefined,
-        requestedCount: "all" | number | undefined): Promise<topology.AvailabilityZoneDescription[]> {
+        requestedZones: VpcArgs["requestedAvailabilityZones"] = 2): Promise<topology.AvailabilityZoneDescription[]> {
 
     const result = await aws.getAvailabilityZones(/*args:*/ undefined, { provider, async: true });
     if (result.names.length !== result.zoneIds.length) {
         throw new pulumi.ResourceError("Availability zones for region had mismatched names and ids.", parent);
     }
 
-    const descriptions: topology.AvailabilityZoneDescription[] = [];
-    for (let i = 0, n = result.names.length; i < n; i++) {
-        descriptions.push({ name: result.names[i], id: result.zoneIds[i] });
-    }
 
-    const count =
-        typeof requestedCount === "number" ? requestedCount :
-        requestedCount === "all" ? descriptions.length : 2;
+    const descriptions = result.names.map((name, idx) => ({ name, id: result.zoneIds[idx] }) )
+
+
+    if (typeof requestedZones === "object") {
+        return new Promise((resolve, reject) => {
+            pulumi.Output.create(requestedZones).apply(requestedZones => {
+                const mappedZones = descriptions.filter(zone => requestedZones.includes(zone.name));
+                mappedZones.length === requestedZones.length ?
+                    resolve(mappedZones) :
+                    reject(new pulumi.ResourceError("Availability zones did not match requested zones", parent));
+            })
+        })
```

Comment:
nit: missing semicolons (the linter in actions is running against master and not this branch)

Reply:
Heh, I had to disable my local formatter and you guys don't got a configuration supplied, I'd make a PR myself, but I am not sure what your specs are :)